modules/userdirectory/src/main/java/org/opencastproject/userdirectory/InMemoryUserAndRoleProvider.java

```diff
@@ -114,19 +121,50 @@ private void createSystemUsers() {
     for (Organization organization : orgDirectoryService.getOrganizations()) {
       JaxbOrganization jaxbOrganization = JaxbOrganization.fromOrganization(organization);
 
-      // Create the digest auth user with a clear text password
+      // Create the digest auth users with clear text passwords
+
+      // Role set for the system user
       Set<JaxbRole> roleList = new HashSet<JaxbRole>();
       for (String roleName : SecurityConstants.GLOBAL_SYSTEM_ROLES) {
         roleList.add(new JaxbRole(roleName, jaxbOrganization));
       }
 
-      // Create the digest user
+      // Role set for the capture agent user
+      Set<JaxbRole> caRoleList = new HashSet<JaxbRole>();
+      for (String roleName : SecurityConstants.GLOBAL_CA_ROLES) {
+        caRoleList.add(new JaxbRole(roleName, jaxbOrganization));
+      }
+
+      // Add the organization anonymous role to the CA user
+      caRoleList.add(new JaxbRole(organization.getAnonymousRole(), jaxbOrganization));
+
+      String caExtraRoles = organization.getProperties().get(CAPTURE_AGENT_EXTRA_ROLES_KEY);
+      String caUsername = organization.getProperties().get(CAPTURE_AGENT_USER_KEY);
+      // Add any extra custom roles to the CA user
+      if ((caExtraRoles != null) && (caUsername != null)) {
+        List<String> items = Arrays.asList(caExtraRoles.split("\\s*,\\s*"));
+        for (String item : items) {
+          logger.debug("Adding custom role '{}' to capture agent user {}", item, caUsername);
+          caRoleList.add(new JaxbRole(item, jaxbOrganization));
+        }
+      }
+
+      // Create the system user
       if (digestUsername != null && digestUserPass != null) {
-        logger.info("Creating the system digest user");
+        logger.info("Creating the system digest user '{}'", digestUsername);
         User digestUser = new JaxbUser(digestUsername, digestUserPass, DIGEST_USER_NAME, null, getName(), true,
                 jaxbOrganization, roleList);
         inMemoryUsers.add(digestUser);
       }
+
+      String caUserPass = organization.getProperties().get(CAPTURE_AGENT_PASSWORD_KEY);
```

Comment:
Can you please:

- Move the logic to create the capture agent user (incl. extra roles) move to one single place
- First check caUsername != null && caUserPass != null and then do the logic to add extra roles